torch/distributed/nn/functional.py

```diff
@@ -318,12 +318,15 @@ def backward(ctx, grad_output):
 class _AllGather(Function):
     @staticmethod
     def forward(ctx, group, tensor):
+        # Need contiguous tensors for collectives.
+        tensor = tensor.contiguous()
+
         ctx.group = group
         out_tensor_list = [
             torch.empty_like(tensor) for _ in range(dist.get_world_size(group=group))
```

Comment:
does this mean empty_like also copies the layout where that's why output_list are non-contiguous?

Reply:
Yes